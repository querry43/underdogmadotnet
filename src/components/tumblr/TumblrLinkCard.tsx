import * as React from 'react'

import TumblrCard from './TumblrCard'

const publisherStyle : React.CSSProperties = {
  color: 'black',
  textDecoration: 'none'
}

const imgStyle : React.CSSProperties = {
  margin: '10px'
}

const excerptStyle : React.CSSProperties = {
  marginTop: '0'
}

const dateStyle : React.CSSProperties = {
  fontSize: 'x-small'
}

class TumblrLinkCard extends TumblrCard {
  protected content() {
    const imgElement = this.props.post.description
      ? <div dangerouslySetInnerHTML={{ __html: this.props.post.description }} />
      : <img style={imgStyle} src={this.props.post.link_image} />

    return (
      <div>
        <a style={publisherStyle} href={this.props.post.url}>
          <p>{this.props.post.publisher}</p>
          {imgElement}
        </a>
        <p style={excerptStyle}>{this.props.post.excerpt}</p>
        <p style={dateStyle}>{new Date(this.props.post.timestamp * 1000).toLocaleString()}</p>
      </div>
    )
  }
}

export default TumblrLinkCard

// {
//   "post": {
//     "type": "text",
//     "blog_name": "qrry43",
//     "blog": {},
//     "id": 179779160723,
//     "post_url": "https://qrry43.tumblr.com/post/179779160723/test-text",
//     "slug": "test-text",
//     "date": "2018-11-05 05:13:20 GMT",
//     "timestamp": 1541394800,
//     "state": "published",
//     "format": "html",
//     "reblog_key": "oZOxjg3Z",
//     "tags": {},
//     "short_url": "https://tmblr.co/ZlGD6Y2dRh4gJ",
//     "summary": "test text",
//     "is_blocks_post_format": false,
//     "recommended_source": null,
//     "recommended_color": null,
//     "note_count": 0,
//     "title": "test text",
//     "body": "<p>\n\ntest text\n\n<br/></p>",
//     "reblog": {},
//     "trail": {},
//     "can_like": false,
//     "can_reblog": false,
//     "can_send_in_message": true,
//     "can_reply": false,
//     "display_avatar": true
//   }
// }
