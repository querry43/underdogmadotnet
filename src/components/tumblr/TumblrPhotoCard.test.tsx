/* tslint:disable:object-literal-sort-keys */

import * as React from 'react'

import TumblrLinkCard from './TumblrPhotoCard'

it('matches snapshot', () => {
  const post = {
    "type": "photo",
    "blog_name": "qrry43",
    "blog": {},
    "id": 179739183803,
    "post_url": "https://qrry43.tumblr.com/post/179739183803",
    "slug": "",
    "date": "2018-11-04 02:32:37 GMT",
    "timestamp": 1541298757,
    "state": "published",
    "format": "html",
    "reblog_key": "ltvXxLHV",
    "tags": {},
    "short_url": "https://tmblr.co/ZlGD6Y2dPIagx",
    "summary": "",
    "is_blocks_post_format": false,
    "recommended_source": null,
    "recommended_color": null,
    "note_count": 0,
    "caption": "",
    "reblog": {},
    "trail": {},
    "image_permalink": "https://qrry43.tumblr.com/image/179739183803",
    "photos": {},
    "can_like": false,
    "can_reblog": false,
    "can_send_in_message": true,
    "can_reply": false,
    "display_avatar": true
  }

  expect(<TumblrLinkCard post={post} />).toMatchSnapshot()
)
