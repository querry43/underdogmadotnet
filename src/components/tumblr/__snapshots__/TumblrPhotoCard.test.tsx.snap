// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<TumblrPhotoCard
  post={
    Object {
      "blog": Object {},
      "blog_name": "qrry43",
      "can_like": false,
      "can_reblog": false,
      "can_reply": false,
      "can_send_in_message": true,
      "caption": "",
      "date": "2018-11-04 02:32:37 GMT",
      "display_avatar": true,
      "format": "html",
      "id": 179739183803,
      "image_permalink": "https://qrry43.tumblr.com/image/179739183803",
      "is_blocks_post_format": false,
      "note_count": 0,
      "photos": Object {},
      "post_url": "https://qrry43.tumblr.com/post/179739183803",
      "reblog": Object {},
      "reblog_key": "ltvXxLHV",
      "recommended_color": null,
      "recommended_source": null,
      "short_url": "https://tmblr.co/ZlGD6Y2dPIagx",
      "slug": "",
      "state": "published",
      "summary": "",
      "tags": Object {},
      "timestamp": 1541298757,
      "trail": Object {},
      "type": "photo",
    }
  }
/>
`;
